src/test/java/cloud/fogbow/ras/core/plugins/interoperability/aws/network/v2/AwsV2NetworkPluginTest.java

```diff
@@ -0,0 +1,616 @@
+import software.amazon.awssdk.services.ec2.model.Route;
+import software.amazon.awssdk.services.ec2.model.RouteTable;
+import software.amazon.awssdk.services.ec2.model.Subnet;
+import software.amazon.awssdk.services.ec2.model.Tag;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({ AwsV2ClientUtil.class, SharedOrderHolders.class })
+public class AwsV2NetworkPluginTest {
+
+	private static final String ANOTHER_VPC_ID = "another-vpc-id";
+	private static final String ANY_VALUE = "anything";
+	private static final String CLOUD_NAME = "amazon";
+	private static final String DEFAULT_CIDR_DESTINATION = "0.0.0.0/0";
+	private static final String FAKE_CIDR_ADDRESS = "1.0.1.0/28";
+	private static final String FAKE_GROUP_ID = "fake-group-id";
+	private static final String FAKE_INSTANCE_NAME = "fake-instance-name";
+	private static final String FAKE_ROUTE_TABLE_ID = "fake-route-table-id";
+	private static final String FAKE_SUBNET_ID = "fake-subnet-id";
+	private static final String FAKE_VPC_ID = "fake-vpc-id";
+	
+	private AwsV2NetworkPlugin plugin;
+	private SharedOrderHolders sharedOrderHolders;
+
+	@Before
+	public void setUp() {
+		String awsConfFilePath = HomeDir.getPath() 
+				+ SystemConstants.CLOUDS_CONFIGURATION_DIRECTORY_NAME
+				+ File.separator 
+				+ CLOUD_NAME 
+				+ File.separator 
+				+ SystemConstants.CLOUD_SPECIFICITY_CONF_FILE_NAME;
+
+		this.plugin = Mockito.spy(new AwsV2NetworkPlugin(awsConfFilePath));
+		this.sharedOrderHolders = Mockito.mock(SharedOrderHolders.class);
+
+		PowerMockito.mockStatic(SharedOrderHolders.class);
+		BDDMockito.given(SharedOrderHolders.getInstance()).willReturn(this.sharedOrderHolders);
+
+		Mockito.when(this.sharedOrderHolders.getOrdersList(Mockito.any(OrderState.class)))
+				.thenReturn(new SynchronizedDoublyLinkedList<>());
+
+		Mockito.when(this.sharedOrderHolders.getActiveOrdersMap()).thenReturn(new HashMap<>());
+	}
+	
+	// test case: When calling the requestInstance method, with a network order and
+	// cloud user valid, a client is invoked to create a network instance, returning
+	// the network ID.
+	@Test
+	public void testRequestInstanceSuccessful() throws FogbowException {
+		// set up
+		Ec2Client client = Mockito.mock(Ec2Client.class);
+		PowerMockito.mockStatic(AwsV2ClientUtil.class);
+		BDDMockito.given(AwsV2ClientUtil.createEc2Client(Mockito.anyString(), Mockito.anyString())).willReturn(client);
+
+		String vpcId = FAKE_VPC_ID;
+		mockCreateSubnets(null, client);
+		mockCreateRouteTables(vpcId, client);
+		mockCreateSecurityGroups(client);
+
+		NetworkOrder networkOrder = Mockito.spy(new NetworkOrder());
+		AwsV2User cloudUser = Mockito.mock(AwsV2User.class);
+
+		String expected = FAKE_SUBNET_ID;
+
+		// exercise
+		String networkId = this.plugin.requestInstance(networkOrder, cloudUser);
+
+		// verify
+		PowerMockito.verifyStatic(AwsV2ClientUtil.class, VerificationModeFactory.times(1));
+		AwsV2ClientUtil.createEc2Client(Mockito.anyString(), Mockito.anyString());
```

Comment:
why do u call it here? Don't u have the client already?

Reply:
This is the test procedure to verify a method using PowerMockito.